helm/nessie/templates/deployment.yaml

```diff
@@ -19,6 +19,9 @@ spec:
       {{- end }}
       labels:
         {{- include "nessie.selectorLabels" . | nindent 8 }}
+        {{- if .Values.podLabels }}
+        {{- toYaml .Values.podLabels | nindent 8 }}
```

Comment:
Don't we want to add these labels also to the `selector.matchLabels` section above?

Reply:
No, typically not. MatchLabels are there to find a pod in Kubernetes, thus they should include the app name, component, release and so on. They are there to find pods that are managed by the deployment.
Take my istio for example: We still want to manage pods with the deployment even if the istio label does not match. In that case, that label should be changed and not a new pod deployed.

Match Labels are typically not editable in the values.yaml.

However, what I did miss: I think it would be good to add those labels to the deployment and the pods. I will add that.

Reply:
Yes, `spec.selector.matchLabels` has to be a subset of `spec.template.metadata.labels`.

> We still want to manage pods with the deployment even if the istio label does not match.

But if you add the custom labels to both places, the labels _will_ match. In fact I cannot think of a case where adding the labels to both places would result in an incorrect deployment. Are you planning to change or delete the labels after the pod is created?

My fear is, if we don't add the custom labels to `spec.selector.matchLabels`, I think it's only a matter of time until somebody opens an issue and asks to add the custom labels there :-)

Reply:
I guess those are edge cases - but for example if you change manually this functional label of a pod, the pod would no longer be managed by the deployment and a new pod would be spun up.

In my opinion matchLabels should not include any functional labels and really only informative labels that are there to identify the pod.
Another option would be OPA / Gatekeeper that might enforce policies and thus change labels. It might change any functional label of a pod.

I am not sure if that counts, but one of my favorite helm charts, the community one from Airflow, does it the same way:
https://github.com/airflow-helm/charts/blob/main/charts/airflow/templates/scheduler/scheduler-deployment.yaml

Its battleproven and trusted by many of companies.